client/src/main/kotlin/no/nav/su/se/bakover/client/oppdrag/avstemming/GrensesnittavstemmingDataBuilder.kt

```diff
@@ -0,0 +1,39 @@
+package no.nav.su.se.bakover.client.oppdrag.avstemming
+
+import no.nav.su.se.bakover.client.oppdrag.toAvstemmingsdatoFormat
+import no.nav.su.se.bakover.domain.oppdrag.avstemming.Avstemming
+import no.nav.su.se.bakover.domain.oppdrag.avstemming.Avstemmingsnøkkel
+
+internal class GrensesnittavstemmingDataBuilder(
+    private val avstemming: Avstemming.Grensesnittavstemming,
+) {
+    private val utbetalinger = avstemming.utbetalinger
+
+    fun build(): GrensesnittsavstemmingData {
+        return GrensesnittsavstemmingData(
+            aksjon = Aksjonsdata.Grensesnittsavstemming(
+                nokkelFom = Avstemmingsnøkkel(avstemming.fraOgMed).toString(),
+                nokkelTom = Avstemmingsnøkkel(avstemming.tilOgMed).toString(),
+                avleverendeAvstemmingId = avstemming.id.toString(),
+            ),
+            total = Totaldata(
+                totalAntall = utbetalinger.size,
+                totalBelop = utbetalinger.flatMap { it.utbetalingslinjer }.sumOf { it.beløp }.toBigDecimal(),
+                fortegn = Fortegn.TILLEGG,
+            ),
+            periode = Periodedata(
+                datoAvstemtFom = avstemming.fraOgMed.toAvstemmingsdatoFormat(),
+                datoAvstemtTom = avstemming.tilOgMed.toAvstemmingsdatoFormat(),
+            ),
+            grunnlag = GrunnlagBuilder(utbetalinger).build(),
+            detalj = DetaljBuilder(utbetalinger).build(),
+        ).also {
+            it.sanityCheck()
+        }
+    }
+
+    private fun GrensesnittsavstemmingData.sanityCheck() {
+        check(total.totalAntall == grunnlag.totaltAntall()) { "Totaldata og grunnlag er uenige om totalt antall utbetalinger!" }
```

Comment:
Hvis dette inntreffer, hvordan vil det påvirke den totale flyten? Vil vi prøve på nytt til det går (dvs. stange i den ene)? Eller vil vi gå videre (og da muligens ignorere denne.)?

Reply:
Misvisende at dette er "grønn" kode (bare nytt navn på eksisterende kode for grensesnittsavstemming) - når det er sagt vil denne hendelsen føre til en kast med påfølgende kræsj, avstemming vil prøves på nytt i hht. autmatisk jobb (neste dag).